File: java/staircase/src/main/java/com/yauritux/StairCase.java
package com.yauritux;

import java.util.Collections;

public class StairCase {

  public static void main(String... args) throws Exception {
    if (args.length < 1) {
      System.err.println("Usage: java StairCase <number of stairs>");
      System.err.println("e.g. : java StairCase 7");
      System.exit(-1);
    }
    int n = 0;
    try {
      n = Integer.parseInt(args[0]);
    } catch (NumberFormatException e) {
      System.err.println("Usage: java StairCase <number of stairs>");
      System.err.println("Invalid Parameter! <number of stairs> should be in numeric");
      System.err.println("java StairCase 7");
      System.exit(-1);
    }
    for (int i = 1; i <= n; i++) {
      System.out.print(String.join("", Collections.nCopies(n - i, " ")));
      System.out.println(String.join("", Collections.nCopies(i, "#")));
    }
  }
}

File: typescript/staircase/Staircase.ts
export default class Staircase {
  printStairs(n: number): string[] {
    let result = [];
    let numberOfSpaces = n - 1;
    let symbol = "*";
    for (let i = 0; i < n - 1; i++) {
      let stairs = "";
      for (let j = 0; j < numberOfSpaces - 1; j++) {
        stairs += " ";
      }
      stairs += symbol;
      result.push(stairs);
      symbol += "*";
      numberOfSpaces--;
    }
    return result;
  }
}
